### ISSUE TITLE
Multipart emails with empty multiline text part break DKIM

### ISSUE THREAD

**Issue by GioPan04** [zone-eu/zone-mta#424]
This is a similar issue to #411, although much specific.
I've noticed that If a multipart email (text and html) is sent and the text part contains more than 5 empty lines (both \r\n and \n) the email won't get correctly signed.

 - This message contains 5 empty lines and gets correctly signed [working.txt](https://github.com/user-attachments/files/17227306/pass.txt)

 - This message contains 6 empty lines and is not correctly signed [fail.txt](https://github.com/user-attachments/files/17227344/fail.txt)

I've also noticed that if a message is not correctly signed one empty line gets removed from the text part (this maybe the reason for the failing DKIM).

 - This is the message downloaded from gmail (the receiving server) which was sent with 5 empty new lines: [pass-gmail.txt](https://github.com/user-attachments/files/17227469/pass-gmail.txt)

 - This is the message downloaded from gmail which was sent with 6 empty new lines: [fail-gmail.txt](https://github.com/user-attachments/files/17227448/fail-gmail.txt)

**Comment by andris9** [comment]
Yes, the current DKIM signing implementation is buggy. I plan to replace it with https://github.com/postalsys/mailauth which has a better, more tested implementation

**Comment by GioPan04** [comment]
Is there any ETA on this?

**Comment by andris9** [comment]
Try [v 3.10.0](https://github.com/zone-eu/zone-mta/releases/tag/v3.10.0)

**Comment by GioPan04** [comment]
I'm still experiencing this issue.
A line between the multiparts is deleted after the dkim is calculated, breaking the previous body hash altogether.

**Comment by GioPan04** [comment]
I was able to isolate the issue.
It seems like that if I connect to TLS or STARTTLS the DKIM is incorrectly calculated. When I connect to the raw 25 port the DKIM is valid.

**Comment by andris9** [comment]
I seem to be unable to reproduce this issue.

I use STARTTLS and send fail.txt and pass.txt to a Gmail address through my ZoneMTA instance. Both messages are signed correctly, and Gmail shows that everything is correct. Do you have, for example, a ZoneMTA plugin that somehow modifes the email and is triggered for the STARTLS interface but not for the cleartext interface?

**Comment by NickOvt** [comment]
Hello!

I tried to send an email using your provided php script and original.txt file contents as the data and indeed the dkim breaks in that case (can't verify body hash). The culprit (highly likely) is the single `=` in the middle of the content. Will investigate and take this issue into work.

**Comment by NickOvt** [comment]
@GioPan04 The issue was very likely due to incorrect message splitting in `MailSplit`'s `MessageSplitter`.
Please view https://github.com/zone-eu/mailsplit/pull/31
You can try to test the email sending with the new splitter. But as far as I have tested it - seems the issue has been resolved.

I'll close this issue when that PR is merged.
